src/test/java/dev/openfeature/sdk/LockingTest.java

```diff
@@ -0,0 +1,117 @@
+package dev.openfeature.sdk;
+
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import dev.openfeature.sdk.internal.AutoCloseableReentrantReadWriteLock;
+
+class LockingTest {
```

Comment:
There wasn't a reasonable means of fully testing the locking with blackbox testing, so I've added this whitebox-style testing.